core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -473,7 +473,7 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
 
     // Validate spark.executor.extraJavaOptions
     getOption(executorOptsKey).foreach { javaOpts =>
-      if (javaOpts.contains("-Dspark")) {
+      if (javaOpts.contains("-Dspark") && !javaOpts.contains("-Dspark.hadoop.")) {
```

Comment:
Thanks for calling this out. @ifilonenko 

FYI, we can actually avoid this change by setting `-Dspark.hadoop.hadoop.security.authenticationi` directly in the command line of the executor Dockerfile without relying on this config key `spark.executor.extraJavaOptions`.  Perhaps that is the right change.